File: Packages/NotebarCore/Sources/NotebarCore/Models/BoardColumn.swift
import Foundation

/// A status column on a `Board` (spec §5's `board_column` table). `kind`
/// drives the one behaviour rule the repository owns — moving a task into a
/// `doneKind` column stamps `completedAt`, moving it out clears it (spec
/// §6.3a) — so it is a string constant rather than a fixed enum, the same
/// reasoning `OpenTab.noteKind` uses: it is schema data a future board
/// could extend, not a closed set the type system should own.
public struct BoardColumn: Identifiable, Equatable, Sendable {
    public var id: String
    public var boardID: String
    public var name: String
    public var kind: String
    public var sortOrder: Double
    public var wipLimit: Int?

    public init(
        id: String = UUID().uuidString,
        boardID: String,
        name: String,
        kind: String,
        sortOrder: Double,
        wipLimit: Int? = nil
    ) {
        self.id = id
        self.boardID = boardID
        self.name = name
        self.kind = kind
        self.sortOrder = sortOrder
        self.wipLimit = wipLimit
    }
}

public extension BoardColumn {
    /// Not-yet-started work. Seeded as "Queue".
    static let backlogKind = "backlog"
    /// In-progress work. Seeded as "Working".
    static let activeKind = "active"
    /// Finished work. Seeded as "Done" — the kind `GRDBTaskRepository`
    /// checks to decide whether a moved task's `completedAt` is stamped or
    /// cleared.
    static let doneKind = "done"
}

File: windows/Notebar.Core/Models/BoardColumn.cs
namespace Notebar.Core.Models;

/// <summary>A status column on a Board.</summary>
/// <remarks>
/// Kind drives the one behaviour rule the repository owns — moving a task into a
/// DoneKind column stamps CompletedAt, moving it out clears it — so it is a
/// string constant rather than a closed enum: it is schema data a future board
/// could extend, not a fixed set the type system should own.
/// </remarks>
public sealed record BoardColumn(
    string Id, string BoardId, string Name, string Kind, double SortOrder, int? WipLimit)
{
    /// <summary>Not-yet-started work. Seeded as "Queue".</summary>
    public const string BacklogKind = "backlog";
    /// <summary>In-progress work. Seeded as "Working".</summary>
    public const string ActiveKind = "active";
    /// <summary>Finished work. Seeded as "Done" — the kind the task repository
    /// checks to decide whether a moved task's CompletedAt is stamped or cleared.</summary>
    public const string DoneKind = "done";
}
